codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpProtocolGeneratorUtils.java

```diff
@@ -60,8 +60,6 @@ private HttpProtocolGeneratorUtils() {}
         writer.openBlock("func $L(response $P) error {", "}", errorFunctionName, responseType, () -> {
             writer.addUseImports(SmithyGoDependency.BYTES);
             writer.addUseImports(SmithyGoDependency.IO);
-            writer.write("defer response.Body.Close()");
```

Comment:
This was just for errors. Do we have to do it for each response deser?

Reply:
The AddErrorCloseResponseBodyMiddleware  helper adds the middleware for all operation responses, and will close response body in case of any response error.

The AddCloseResponseBodyMiddleware is only used for successful responses (which is also why it is only applied to API operations without streaming responses.

Reply:
Still won't we have to remove the explicit response body close statements used by the deserializers?

Reply:
I thought i removed all of them as well, but i'll got back and check to see if i missed any.